dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java

```diff
@@ -135,127 +134,146 @@ private String extractMessage(RoutedMessage routedMessage, LogRecord logRecord)
 
     public LogTraceData parse(RoutedMessage routedMessage) {
 
-        GenericData genData = new GenericData();
+        LogTraceData logData = new LogTraceData();
         LogRecord logRecord = routedMessage.getLogRecord();
-        String messageVal = extractMessage(routedMessage, logRecord);
 
         long dateVal = logRecord.getMillis();
-        genData.addPair(LogFieldConstants.IBM_DATETIME, dateVal);
+        logData.setDatetime(dateVal);
 
         String messageIdVal = null;
-
+        String messageVal = extractMessage(routedMessage, logRecord);
         if (messageVal != null) {
             messageIdVal = parseMessageId(messageVal);
         }
-
-        genData.addPair(LogFieldConstants.IBM_MESSAGEID, messageIdVal);
-
-        int threadIdVal = (int) Thread.currentThread().getId();//logRecord.getThreadID();
-        genData.addPair(LogFieldConstants.IBM_THREADID, threadIdVal);
-        genData.addPair(LogFieldConstants.MODULE, logRecord.getLoggerName());
-        genData.addPair(LogFieldConstants.SEVERITY, LogFormatUtils.mapLevelToType(logRecord));
-        genData.addPair(LogFieldConstants.LOGLEVEL, LogFormatUtils.mapLevelToRawType(logRecord));
-        genData.addPair(LogFieldConstants.IBM_METHODNAME, logRecord.getSourceMethodName());
-        genData.addPair(LogFieldConstants.IBM_CLASSNAME, logRecord.getSourceClassName());
-
-        genData.addPair(LogFieldConstants.LEVELVALUE, logRecord.getLevel().intValue());
+        logData.setMessageId(messageIdVal);
+
+        int threadIdVal = (int) Thread.currentThread().getId();
+        logData.setThreadId(threadIdVal);
+        logData.setModule(logRecord.getLoggerName());
+        logData.setSeverity(LogFormatUtils.mapLevelToType(logRecord));
+        logData.setLoglevel(LogFormatUtils.mapLevelToRawType(logRecord));
+        logData.setMethodName(logRecord.getSourceMethodName());
+        logData.setClassName(logRecord.getSourceClassName());
+        logData.setLevelValue(logRecord.getLevel().intValue());
         String threadName = Thread.currentThread().getName();
-        genData.addPair(LogFieldConstants.THREADNAME, threadName);
+        logData.setThreadName(threadName);
 
         WsLogRecord wsLogRecord = getWsLogRecord(logRecord);
-
         if (wsLogRecord != null) {
-            genData.addPair(LogFieldConstants.CORRELATION_ID, wsLogRecord.getCorrelationId());
-            genData.addPair(LogFieldConstants.ORG, wsLogRecord.getOrganization());
-            genData.addPair(LogFieldConstants.PRODUCT, wsLogRecord.getProduct());
-            genData.addPair(LogFieldConstants.COMPONENT, wsLogRecord.getComponent());
+            logData.setCorrelationId(wsLogRecord.getCorrelationId());
+            logData.setOrg(wsLogRecord.getOrganization());
+            logData.setProduct(wsLogRecord.getProduct());
+            logData.setComponent(wsLogRecord.getComponent());
+        } else {
```

Comment:
We need them to be null because if I don't add set them to null, it'll throw a NullPointerException when it reaches setSequence. This is because the last set method setComponent stored a value at index 9. When it reaches setSequence, setSequence will try to store a value at index 14. This will throw a NullPointerException because we can only add to ArrayLists incrementally.

Reply:
ok, that's pointing to ArrayList being the wrong approach for GenericData, but fixing that should probably be a separate issue.